### DISCUSSION TITLE
Database size increases for no apparent reason

### DISCUSSION THREAD

**Discussion by Phil632** [supabase/supabase#22198] [Questions] [answered]
I built a project using prisma and deployed it serverless on AWS. In this setup my database size increases exponentially for no apparent reason. Yesterday I dumped my database tables to a new supabase project (because the db size had reached 6GB). Within a day the db size has increased from the starting 75MB to 650MB (I have not interacted with the database in the meantime). I have no idea why this is happening. Following the [documentation](https://supabase.com/partners/integrations/prisma) and my project's connection info, I interact with the database through these connection strings:

Connect to Supabase via connection pooling with Supavisor.
DATABASE_URL="postgres://postgres.[MY-DATABASE]:[MY-PASSWORD]@aws-0-eu-central-1.pooler.supabase.com:6543/postgres?pgbouncer=true&connection_limit=1"

Direct connection to the database. Used for migrations.
DIRECT_URL="postgres://postgres.[MY-DATABASE]:[MY-PASSWORD]@aws-0-eu-central-1.pooler.supabase.com:5432/postgres"

Does anyone know why this is happening? Any help is apreciated.

**Comment by GaryAustin1** [comment, marked as answer]
Have you looked in the logs for api activity?   Some cron task you have running? If you have two instances doing this seems on your side.

**Reply by GaryAustin1** [reply]
Yeah that rate of connections from Prisma does not look good.  I've got nothing for you though on Prisma, but clearly you need to stop that.

**Reply by Phil632** [reply]
Haha yeah, I'll look into that. Thanks for pointing me in the right direction. No idea why it never occurred to me to look at the logs lol.

**Comment by Phil632** [comment]
Seems like prisma keeps trying to connect to a shadow database that does not exist.
![Screenshot 2024-03-24 at 5 06 03 PM](https://github.com/supabase/supabase/assets/52404694/a31c588f-abce-4988-bc27-d6689b85aec9)
